From 2afd5763c5a7ff8f6f853903c26d04fec92b5e4c Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 6 Jun 2019 15:58:23 +0530
Subject: [PATCH 047/138] octeontx2-pf: Compile PF driver only when enabled

commit 82fdc3e40517e4a1c8b44dc3a7bff9af5e41c4cd from
git@git.assembla.com:cavium/WindRiver.linux.git

Compile PF driver only when enabled in config

Change-Id: I827b5066646dbb7ffa28dc98b5911f9cd526e8d6
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10587
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/Kconfig  | 1 +
 drivers/net/ethernet/marvell/octeontx2/Makefile | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/Kconfig b/drivers/net/ethernet/marvell/octeontx2/Kconfig
index 0711767d95ae..11e1368db10d 100644
--- a/drivers/net/ethernet/marvell/octeontx2/Kconfig
+++ b/drivers/net/ethernet/marvell/octeontx2/Kconfig
@@ -27,6 +27,7 @@ config NDC_DIS_DYNAMIC_CACHING
 
 config OCTEONTX2_PF
 	tristate "Marvell OcteonTX2 NIC Physical Function driver"
+	select OCTEONTX2_MBOX
 	depends on PCI && ARM64 && ARM64_LSE_ATOMICS
 	help
 	  This driver supports Marvell's OcteonTX2 Resource Virtualization
diff --git a/drivers/net/ethernet/marvell/octeontx2/Makefile b/drivers/net/ethernet/marvell/octeontx2/Makefile
index 563dc9cd6b36..0064a69e0f72 100644
--- a/drivers/net/ethernet/marvell/octeontx2/Makefile
+++ b/drivers/net/ethernet/marvell/octeontx2/Makefile
@@ -5,4 +5,4 @@
 
 obj-$(CONFIG_OCTEONTX2_MBOX) += af/
 obj-$(CONFIG_OCTEONTX2_AF) += af/
-obj-y			   += nic/
+obj-$(CONFIG_OCTEONTX2_PF) += nic/
-- 
2.17.1

